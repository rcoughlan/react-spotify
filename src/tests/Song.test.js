import Album from '../components/Album';
import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

describe('Artist Component', () => {

    let artistObject = {
        artist: 'Arctic Monkeys',
        genres: ['rock', 'indie', 'sheffield indie'],
        artistImage: 'https://i.scdn.co/image/ed0552e9746ed2bbf04ae4bcb5525700ca31522d',
        url: 'https://open.spotify.com/artist/7Ln80lUS6He07XvHI8qqHH'

    }

    it('matches the snapshot', () => {
        const tree = renderer.create(<Artist artist={artistObject.artist} genres={artistObject.genres} artistImage={artistObject.artistImage} url={artistObject.url}/>).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('should render a component with a thumbnail image with correct alt tag', () => {
        const wrapper = shallow(<Songs artist={artistObject.artist} genres={artistObject.genres} artistImage={artistObject.artistImage} url={artistObject.url}/>);
        const thumbnail = wrapper.find('img.thumbnail');

        expect(thumbnail.prop('alt')).toBe(artistObject.artist + ' artist image');
        expect(thumbnail.prop('src')).toBe(artistObject.artistImage);
    });
    
    it('thumbnail image should link to the artist on spotify', () => {
        const wrapper = shallow(<Songs artist={artistObject.artist} genres={artistObject.genres} artistImage={artistObject.artistImage} url={artistObject.url}/>);
        const thumbnail = wrapper.find('img.thumbnail');

        expect(thumbnail.prop('href')).toBe(artistObject.url);
        expect(thumbnail.prop('target')).toBe('_blank');
    });
    
    it('should have an artist title', () => {
        const wrapper = shallow(<Songs artist={artistObject.artist} genres={artistObject.genres} artistImage={artistObject.artistImage} url={artistObject.url}/>);
        const artist = wrapper.find('p.artist');
        expect(artist.prop('alt')).toBe('artist: ' + artistObject.artist);
    });
    
    it('should have an list of genres', () => {
        const wrapper = shallow(<Songs artist={artistObject.artist} genres={artistObject.genres} artistImage={artistObject.artistImage} url={artistObject.url}/>);
        const album = wrapper.find('p.genres');
        expect(album.prop('alt')).toBe('genres: ' + artistObject.genres);
    });
});