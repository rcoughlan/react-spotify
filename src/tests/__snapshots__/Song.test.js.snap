// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Song Component matches the snapshot 1`] = `
<div
  className="col"
>
  <a
    className="spotify-link"
    href="https://open.spotify.com/album/5ceB3rxgXqIRpsOvVzTG28"
    target="_blank"
  >
    <div
      className="music-card card"
    >
      <img
        alt="Aquemini album art"
        className="card-img-top thumbnail"
        src="https://i.scdn.co/image/931c3a43c71a4f2b86653d78050af585029e5623"
      />
      <div
        className="card-body"
      >
        <div
          alt="track title: undefined"
          className="track-title card-title h5"
        />
        <p
          alt="artist: Outkast"
          className="track-artist card-text"
        >
          Outkast
        </p>
        <p
          alt="album title: Aquemini"
          className="album-title card-text"
        >
          Aquemini
        </p>
      </div>
    </div>
  </a>
</div>
`;
